pkg/webcas/webcas.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package webcas
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/trustbloc/edge-core/pkg/log"
+	"github.com/trustbloc/sidetree-core-go/pkg/api/cas"
+	"github.com/trustbloc/sidetree-core-go/pkg/restapi/common"
+)
+
+const cidPathVariable = "cid"
+
+type logger interface {
+	Errorf(msg string, args ...interface{})
+}
+
+// WebCAS represents a WebCAS handler + client for the backing CAS.
+type WebCAS struct {
+	casClient cas.Client
+	logger    logger
+}
+
+// Path returns the HTTP REST endpoint for the WebCAS service.
+func (w *WebCAS) Path() string {
+	return fmt.Sprintf("/cas/{%s}", cidPathVariable)
+}
+
+// Method returns the HTTP REST method for the WebCAS service.
+func (w *WebCAS) Method() string {
+	return http.MethodGet
+}
+
+// Handler returns the HTTP REST handler for the WebCAS service.
+func (w *WebCAS) Handler() common.HTTPRequestHandler {
+	return w.handler
+}
+
+// New returns a new WebCAS, which contains a REST handler that implements WebCAS as defined in
+// https://trustbloc.github.io/did-method-orb/#webcas.
+func New(casClient cas.Client) *WebCAS {
+	return &WebCAS{casClient: casClient, logger: log.New("webcas")}
+}
+
+func (w *WebCAS) handler(rw http.ResponseWriter, req *http.Request) {
+	cid := mux.Vars(req)[cidPathVariable]
+
+	content, err := w.casClient.Read(cid)
+	if err != nil {
+		rw.WriteHeader(http.StatusNotFound)
```

Comment:
Originally I was going to split this into two scenarios: `StatusNotFound` and `InternalServerError` depending on the error returned from the CAS client. The problem is that I don't think I have a reliable way of detecting this based on the error returned by `w.casClient.Read`.

If `casClient` is the new CAS I implemented in my last PR, then I can check using errors.Is(err, ariesspi.ErrDataNotFound) since all Aries storage implementation will return that error... but if the IPFS client is used, then the error I get back is different and seems to depend on a timeout rather than a more firm "not found" type of message: `Post "http://ipfs:5001/api/v0/cat?arg=QmeKWPxUJP9M3WJgBuj8ykLtGU37iqur5gZ8cDCi49WJVG": context deadline exceeded (Client.Timeout exceeded while awaiting headers)`... so I'm not sure if it's a good idea/reliable to be hardcoding a string check for that.

So currently I just have it returning `StatusNotFound` for any error from the CAS client. Let me know what you think of this.

Reply:
You should not worry about IPFS client, I would focus on doing our implementation properly. We can modify IPFS client to catch "context deadline exceeded"  and return StatusNotFound.

Reply:
@sandrask So right now https://trustbloc.github.io/did-method-orb/#webcas only specified what a success response looks like - it doesn't specify what sorts of error responses are expected, so I'm not sure what status codes exactly should be returned - I'm assuming StatusNotFound is one of them. But are you saying that we should have separate `StatusNotFound` and `InternalServerError` returns?

Reply:
OK, I've updated my PR to distinguish between the two. For now, I put the common error that I check for in `cas.go` - I wasn't really sure where to put it. @sandrask @bstasyszyn @fqutishat Can you take another look at my PR?